### PR TITLE
ft(cyclops-ctrl): Added support for fetching Secret resources

### LINKED ISSUE #271: Implement resource details for Secrets
**Is your feature request related to a problem? Please describe.**
When fetching resources for a Module, some resources are supported to return details about a specific resource. For example, the endpoint for Deployments will return more information about pods and their statuses, and the endpoint for Services will return information about the ports.

If you want to fetch information about a Deployment form default namespace called demo-app you would call:
/resources?group=apps&version=v1&kind=Deployment&name=demo-app&namespace=default

The goal of the issue is to support Secrets on the same endpoint. The endpoint for fetching Secret data would look like this: /resources?group=&version=v1&kind=Secret&name=<name>&namespace=<namespace>

The handler function for the endpoint is [here](https://github.com/cyclops-ui/cyclops/blob/a648a84971c3ffaed600d20159618a55a62c31ea/cyclops-ctrl/internal/controller/modules.go#L516), and you will have to handle a new type [here](https://github.com/cyclops-ui/cyclops/blob/cdb675623c58e4389c9b1abeb9b041d024e6ed7f/cyclops-ctrl/internal/cluster/k8sclient/client.go#L212).

You can add all the keys of the Secret in the response but not the values. We don't want to show the values since they are secret.

Refer to this PR on adding new types #259

### PR DESCRIPTION
**Pull request by siddhantprateek**
This resolves
closes #271

### Endpoint:
```
GET /resources?group=&version=v1&kind=Secret&name=<secret-name>&namespace=<namespace>
```

Implementation result
```
GET { HOST }/resources?group=&version=v1&kind=Secret&name=justsecret-2&namespace=cyclops
```
Output Response:
```json
{
    "group": "",
    "version": "v1",
    "kind": "Secret",
    "name": "justsecret-2",
    "namespace": "cyclops",
    "type": "Opaque",
    "data": {
        "KEY": "VkFMVUU="
    },
    "deleted": false
}
```

## ✅ Checks
- [ ] I have updated the documentation as required
- [x] I have performed a self-review of my code

### CODE REVIEW

File: cyclops-ctrl/internal/models/dto/k8s.go [hunk 1]
```diff
@@ -291,6 +291,57 @@ func (p *PersistentVolumeClaim) SetDeleted(deleted bool) {
...
 }
 
+type Secret struct {
+	Group     string            `json:"group"`
+	Version   string            `json:"version"`
+	Kind      string            `json:"kind"`
+	Name      string            `json:"name"`
+	Namespace string            `json:"namespace"`
+	Type      string            `json:"type"`
+	Data      map[string][]byte `json:"data"`
```
Comment (petar-cvit) [inline cyclops-ctrl/internal/models/dto/k8s.go:301]: Can you send only the keys from the secret without values? I would like to allow users to check which secret keys can they use, but without exposing the secret value

Comment (siddhantprateek) [reply cyclops-ctrl/internal/models/dto/k8s.go:301]: Cool, is this the response you are expecting?
```json
{
    "group": "",
    "version": "v1",
    "kind": "Secret",
    "name": "justsecret-2",
    "namespace": "cyclops",
    "type": "Opaque",
    "dataKeys": [
        "KEY"
    ],
    "deleted": false
}
```

Comment (petar-cvit) [reply cyclops-ctrl/internal/models/dto/k8s.go:301]: Yeah, looks good

**Review by petar-cvit — APPROVED** [review]
Thanks @siddhantprateek

**Pull request merged**